src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -1333,7 +1333,7 @@ public void cacheKey(DecoratedKey key, RowIndexEntry info)
     {
         CachingParams caching = metadata().params.caching;
 
-        if (!caching.cacheKeys() || keyCache == null || keyCache.getCapacity() == 0)
+        if (!caching.cacheKeys() || keyCache == null || !keyCache.enabled())
```

Comment:
if we made "enabled" static and its capacity is set in runtime to 0 then it would not return here, would it? So it means that we would continue with the logic while we could work with a cache where capacity is 0. That does not make sense to me ...

Reply:
Yes, but I doubt this is a use case - the behaviour is not tested as far as I can see (the enabled cache with setting capacity to zero). For some of the cases setting to zero will throw an exception. 

If we want to make it runtime configurable then we should rely on DD and push the change to all the branches we have.

Reply:
We can also catch all the `setCapacity(0)` method calls and set `enabled = false`.

Reply:
Example:
https://github.com/apache/cassandra/pull/3644/commits/6be8b3e330ae1af40efe110ca646bb0ccb08f9c2

Reply:
@Mmuzaf I think if somebody wants to disable cache in runtime then they set the size to 0. I know that it is most probably not documented but I would not be surprised if this pattern was recognized and used widely.